ig "panic: getting host net ns inode" when /host not available

## Description

Running `ig` with `kubectl debug node` with the netadmin profile does not prepare the /host mount. Instead of returning an error, it panics.

## Impact

User is confused about what to do to solve the misconfiguration.

## Environment and steps to reproduce

```
$ kubectl debug --profile=netadmin node/controlplane -ti --image=ghcr.io/inspektor-gadget/ig -- ig trace exec --host --auto-mount-filesystems
Creating debugging pod node-debugger-controlplane-2v4xm with container debugger on node controlplane.
panic: getting host net ns inode: stat /host/proc/1/ns/net: no such file or directory

goroutine 1 [running]:
github.com/inspektor-gadget/inspektor-gadget/pkg/container-collection.WithLinuxNamespaceEnrichment()
        /go/src/github.com/inspektor-gadget/inspektor-gadget/pkg/container-collection/options.go:668 +0xaa
github.com/inspektor-gadget/inspektor-gadget/pkg/ig-manager.NewManager({0xc00058b1e0?, 0x4, 0x4})
        /go/src/github.com/inspektor-gadget/inspektor-gadget/pkg/ig-manager/ig-manager.go:98 +0x1d0
github.com/inspektor-gadget/inspektor-gadget/pkg/operators/localmanager.(*LocalManager).Init(0xc00047a5c0, 0xc000315788)
        /go/src/github.com/inspektor-gadget/inspektor-gadget/pkg/operators/localmanager/localmanager.go:194 +0x1a7
github.com/inspektor-gadget/inspektor-gadget/pkg/operators.(*operatorWrapper).Init.func1()
        /go/src/github.com/inspektor-gadget/inspektor-gadget/pkg/operators/operators.go:130 +0x37
sync.(*Once).doSlow(0xe9efc5?, 0xc0005df600?)
        /usr/local/go/src/sync/once.go:74 +0xc2
sync.(*Once).Do(...)
        /usr/local/go/src/sync/once.go:65
github.com/inspektor-gadget/inspektor-gadget/pkg/operators.(*operatorWrapper).Init(0x21a7060?, 0xc000120c90?)
        /go/src/github.com/inspektor-gadget/inspektor-gadget/pkg/operators/operators.go:129 +0x5d
github.com/inspektor-gadget/inspektor-gadget/pkg/operators.Operators.Init({0xc00048a000?, 0x1, 0x2778a38?}, 0x6?)
        /go/src/github.com/inspektor-gadget/inspektor-gadget/pkg/operators/operators.go:188 +0xaf
github.com/inspektor-gadget/inspektor-gadget/cmd/common.buildCommandFromGadget.func2(0xc0005dfdc8?, {0xc00058b000?, 0x7fff9f6e7e0a?, 0x2?})
        /go/src/github.com/inspektor-gadget/inspektor-gadget/cmd/common/registry.go:269 +0x4b5
github.com/spf13/cobra.(*Command).execute(0xc00042ac00, {0xc00058b000, 0x2, 0x2})
        /go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:940 +0x862
github.com/spf13/cobra.(*Command).ExecuteC(0xc000354300)
        /go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:1068 +0x3bd
github.com/spf13/cobra.(*Command).Execute(...)
        /go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:992
main.main()
        /go/src/github.com/inspektor-gadget/inspektor-gadget/cmd/ig/main.go:64 +0x1a5
```

## Expected behavior

The `pkg/utils/host` package detects the issue early and returns a proper error instead of panic().

--- a/pkg/container-collection/options.go
+++ b/pkg/container-collection/options.go
@@ -664,14 +664,14 @@ func WithCgroupEnrichment() ContainerCollectionOption {
 
 // WithLinuxNamespaceEnrichment enables an enricher to add the namespaces metadata
 func WithLinuxNamespaceEnrichment() ContainerCollectionOption {
-	// GetNetNs() needs a pid in the host pid namespace: it uses $HOST_ROOT/proc/$pid/ns/net
-	// This needs CAP_SYS_PTRACE.
-	netnsHost, err := containerutils.GetNetNs(1)
-	if err != nil {
-		panic(fmt.Sprintf("getting host net ns inode: %s", err))
-	}
-
 	return func(cc *ContainerCollection) error {
+		// GetNetNs() needs a pid in the host pid namespace: it uses $HOST_ROOT/proc/$pid/ns/net
+		// This needs CAP_SYS_PTRACE.
+		netnsHost, err := containerutils.GetNetNs(1)
+		if err != nil {
+			return fmt.Errorf("getting host net ns inode: %w", err)
+		}
+
 		cc.containerEnrichers = append(cc.containerEnrichers, func(container *Container) bool {
 			pid := int(container.Pid)
 			if pid == 0 {

--- a/pkg/utils/host/namespaces.go
+++ b/pkg/utils/host/namespaces.go
@@ -59,7 +59,7 @@ func IsHostNetNs() (bool, error) {
 func isHostNamespace(nsKind string) (bool, error) {
 	if !initDone {
 		// HostProcFs can be overwritten by workarounds, so Init() must be called first.
-		panic("host.Init() must be called before calling isHostNamespace()")
+		return false, fmt.Errorf("host.Init() must be called before calling isHostNamespace()")
 	}
 
 	selfFileInfo, err := os.Stat("/proc/self/ns/" + nsKind)

